src/org/anjocaido/groupmanager/utils/Tasks.java

```diff
@@ -183,54 +182,54 @@ public static String getStringListInString(List<String> list) {
 		if (list == null) {
 			return "";
 		}
-		String result = "";
+		StringBuilder result = new StringBuilder();
 		for (int i = 0; i < list.size(); i++) {
-			result += list.get(i);
+			result.append(list.get(i));
 			if (i < list.size() - 1) {
-				result += ", ";
+				result.append(", ");
 			}
 		}
-		return result;
+		return result.toString();
 	}
 
 	public static String getStringArrayInString(String[] list) {
 
 		if (list == null) {
 			return "";
 		}
-		String result = "";
+		StringBuilder result = new StringBuilder();
 		for (int i = 0; i < list.length; i++) {
-			result += list[i];
+			result.append(list[i]);
 			if (i < ((list.length) - 1)) {
-				result += ", ";
+				result.append(", ");
 			}
 		}
-		return result;
+		return result.toString();
 	}
 
 	public static String getGroupListInString(List<Group> list) {
 
 		if (list == null) {
 			return "";
 		}
-		String result = "";
+		StringBuilder result = new StringBuilder();
 		for (int i = 0; i < list.size(); i++) {
-			result += list.get(i).getName();
+			result.append(list.get(i).getName());
 			if (i < list.size() - 1) {
-				result += ", ";
+				result.append(", ");
 			}
 		}
-		return result;
+		return result.toString();
 	}
 
 	public static String join(String[] arr, String separator) {
 
 		if (arr.length == 0)
 			return "";
-		String out = arr[0];
+		StringBuilder out = Optional.ofNullable(arr[0]).map(StringBuilder::new).orElse(null);
 		for (int i = 1; i < arr.length; i++)
-			out += separator + arr[i];
-		return out;
+			out = (out == null ? new StringBuilder("null") : out).append(separator).append(arr[i]);
+		return out == null ? null : out.toString();

```

Comment:
This is overly complex for a simple method. I really dislike Optional.
StringBuilder yes, Optional is excessive. I prefer simple readable code.

Reply:
I guess it used Optional because I used null-safe when converting to a StringBuilder, will fix